src/sentry/models/event.py

```diff
@@ -476,13 +476,60 @@ def id(self):
         # have to reference the row id anyway.
         return self.event_id
 
-    @property
-    def next_event(self):
-        return None
+    def next_event_id(self, environments=[]):
+        from sentry.utils import snuba
 
-    @property
-    def prev_event(self):
-        return None
+        conditions = [
+            ["timestamp", ">=", self.timestamp],
+            [["timestamp", ">", self.timestamp], ["event_id", ">", self.event_id]]
+        ]
+
+        if len(environments) > 0:
+            conditions.append(['environment', 'IN', environments])
+
+        result = snuba.raw_query(
+            start=self.datetime,  # gte current event
+            end=datetime.utcnow(),  # will be clamped to project retention
+            selected_columns=['event_id'],
+            conditions=conditions,
+            filter_keys={
+                'project_id': [self.project_id],
+                'issue': [self.group_id],
+            },
+            orderby=['timestamp', 'event_id'],
+            limit=1
+        )
+
+        if 'error' in result or len(result['data']) == 0:
+            return None
+
+        return six.text_type(result['data'][0]['event_id'])
+
+    def prev_event_id(self, environments=None):
+        from sentry.utils import snuba
+
+        conditions = [['event_id', '!=', self.id]]
```

Comment:
these should be the same 3 conditions as `next_event_id` but with `<`s instead of `>`s